lnclient/lnd/lnd.go

```diff
@@ -628,6 +628,68 @@ func (svc *LNDService) MakeInvoice(ctx context.Context, amount int64, descriptio
 	for _, channel := range channels {
 		if channel.Active && channel.Public {
 			hasPublicChannels = true
+			break
+		}
+	}
+
+	var hints []*lnrpc.RouteHint
+	if !hasPublicChannels && throughNodePubkey != nil {
```

Comment:
So we don't require mentioning the route hint if the node is public? (for rebalancing that is, I'm not sure how that works)

Reply:
Currently if the node has at least one public channel then we only expose the public channels (existing functionality to match how LDK works)

Reply:
Yeah, even before that we have the much more common usecase that if the user has public channels then they cannot receive at all on their private channels